components/gitlab/new-branch.js

```diff
@@ -0,0 +1,68 @@
+const gitlab = require("https://github.com/PipedreamHQ/pipedream/components/gitlab/gitlab.app.js");
+
+module.exports = {
+  name: "New Branch (Instant)",
+  description: "Emits an event when a new branch is created",
+  version: "0.0.1",
+  props: {
+    gitlab,
+    projectId: { propDefinition: [gitlab, "projectId"] },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  hooks: {
+    async activate() {
+      const hookParams = {
+        push_events: true,
+        url: this.http.endpoint,
+      };
+      const opts = {
+        hookParams,
+        projectId: this.projectId,
+      };
+      const response = await this.gitlab.createHook(opts);
+
+      const { hookId, token } = response;
+      this.db.set("hookId", hookId);
+      this.db.set("token", token);
+    },
+    async deactivate() {
+      const hookId = this.db.get("hookId");
+      const opts = {
+        hookId,
+        projectId: this.projectId,
+      };
+      this.gitlab.deleteHook(opts);
```

Comment:
Do you need to `await` this?

Reply:
Short answer: yes, I think I switched branches and the old `await` got lost.